Add search modifiers in Search | Fix #7967

charanjit-singh opened the pull request:
Fixes #7967 

### Summary of Changes

* Added Regex, Case Sensitive, Whole word modifiers in search ( Ctrl + shift + F )
* Added styles for active classes on modifier buttons.

### Test Plan

Tests can be done by searching using combination of all modifiers and checking their results.

Example test plan:

- [x] Open test file in browser.
- [x] Press Ctrl + Shift + F for searching.
- [x] In Input box, type the words to test.
- [x] Check/ uncheck the modifiers.
- [x] Verify the expected results.


### Screenshots
![screenshot from 2019-02-26 17-24-15](https://user-images.githubusercontent.com/23359602/53416549-beaa9100-39f9-11e9-83a9-5858b4eb5cd5.png)
---

![screenshot from 2019-02-26 17-24-23](https://user-images.githubusercontent.com/23359602/53416556-c0745480-39f9-11e9-8969-167c072327dd.png)

---

![screenshot from 2019-02-26 17-24-44](https://user-images.githubusercontent.com/23359602/53416563-c4a07200-39f9-11e9-8ce4-56fb5e8b1ccb.png)

charanjit-singh commented:
Tests are failing as we need to update tests also.

darkwing commented:
Thank you for submitting @charanjit-singh!  Quick note:  Jest tests are failing because of old snapshots; you can run `yarn jest -u` to update those and add as a second commit! 👍

darkwing commented on src/utils/sources-tree/sortTree.js:
```diff
@@ -18,6 +18,7 @@ export function sortEntireTree(
   tree: TreeDirectory,
   debuggeeUrl: string = ""
 ): TreeDirectory {
+  console.log("Sorting ENtire Tree;");
```
Ooops, I don't think we want this. :)

charanjit-singh replied:
Oh sorry

darkwing commented:
Functionally this works great!  One quick style nit:

When a button is active (or turned on) for find in file, the icon is blue and shows a light hover color; in your PR, the icon doesn't change color and the highlighted color is darker.  Let's do the same styling as the find in file.

This is really impressive @charanjit-singh !  Great work!

charanjit-singh commented:
Okay I'll update it soon.
Thanks for reviewing.

charanjit-singh commented:
@darkwing  I'm unable to update tests using `yarn jest -u`.  Help !

darkwing commented:
Hello @charanjit-singh!  I can update jest tests but I need you to check out a few of the test failures -- it looks like results may be being lost

charanjit-singh commented:
Hi !, 
There was a problem. 
Existing `ProjectSearch` was not in Redux, That's why, The way, I was implementing was failing tests.
So, I shifted back to `master`, added `redux` and fixed above issue.
 Now I'm  creating a new pull request from new branch. Kindly Delete this Pull request.

charanjit-singh commented:
Pull Request #8051
